pkg/p2p/libp2p/headers_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package libp2p_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/p2p"
+	"github.com/ethersphere/bee/pkg/p2p/libp2p"
+)
+
+func TestHeaders(t *testing.T) {
+	headers := p2p.Headers{
+		"test-header-key": []byte("header-value"),
+		"other-key":       []byte("other-value"),
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	s1, overlay1, cleanup1 := newService(t, libp2p.Options{})
+	defer cleanup1()
+
+	s2, _, cleanup2 := newService(t, libp2p.Options{})
+	defer cleanup2()
+
+	var gotHeaders p2p.Headers
+	handled := make(chan struct{})
+	if err := s1.AddProtocol(newTestProtocol(func(_ context.Context, _ p2p.Peer, stream p2p.Stream) error {
+		gotHeaders = stream.Headers()
```

Comment:
Maybe it would be good to test if the input params are good in this kind of function? What do you think?

Reply:
You refer to the anonymous handler func here? That would be good to have, also. Will add checks on input parameters.

Reply:
Exactly, basically, just to check if the parameters received in function are good, if possible. So we know that it is called with right parameters.